### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by mpaneghel** [yarnpkg/berry#1327]
- [ ] I'd be willing to implement a fix

**Describe the bug**

When setting a custom registry server with https (and no auth), yarn install hangs at 
```
YN0000: ┌ Resolution step
```
`yarn add` also hangs, and presumably any command that needs communication with the registry.

**To Reproduce**

- Link to a custom registry server in .yarnrc.yml, this is how mine looks like:

```
yarnPath: .yarn/releases/yarn-2.0.0-rc.33.js
npmRegistryServer: https://my-registry.com
```

- Try adding any package or installing existing ones.

**Environment if relevant (please complete the following information):**

 - OS: macOS
 - Node version: 10.16.3
 - Yarn version: 2.0.0-rc.33

Thanks!

**Comment by paul-soporan** [comment]
Duplicate of #1284. The fix already landed in `master`, so you can do [`yarn set version from sources`](https://yarnpkg.com/cli/set/version/from/sources). 🙂

**Comment by mpaneghel** [comment]
Thanks so much for the help! I managed to enable the timeout through `yarn set version from sources`.

I'm reopening this as I now can see the actual error:
```
YN0001: │ RequestError: ENOTFOUND my-registry.com
    at ClientRequest.<anonymous> (/Users/mpaneghel/Projects/test/.yarn/releases/yarn-sources.js:10:287248)
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:327:22)
    at ClientRequest.e.emit (/Users/mpaneghel/Projects/test/.yarn/releases/yarn-sources.js:16:317084)
    at TLSSocket.socketErrorListener (_http_client.js:463:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:100:8)
    at emitErrorCloseNT (internal/streams/destroy.js:68:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
    at d.lookupAsync (/Users/mpaneghel/Projects/test/.yarn/releases/yarn-sources.js:10:398199)
```
The root cause seems to be a bug in `cacheable-lookup`, one of the dependencies of `got`. `yarnpkg-core` uses `got@^11.0.0-beta.1` which, in turn, had `cacheable-lookup@^4.1.1` as one of its dependencies. This version has the problem mentioned above.

Interestingly, if I pull `got@^11.0.0-beta.1` locally, the `cacheable-lookup` dependency is resolved to `4.3.0`, which doesn't seem to show the problem.

Maybe the `got` version could be installed again with the correct version of `cacheable-lookup`, or a package extension could be added for the `got` package, which I saw in .yarnrc already has some extensions declared?

Thanks again for the help 🙂
